src/gui/mainwindow.cpp

```diff
@@ -1939,12 +1939,11 @@ void MainWindow::handleUpdateCheckFinished(const bool updateAvailable, const QSt
             , tr("No updates available.\nYou are already using the latest version.")
             , QMessageBox::Ok, this};
         msgBox->setAttribute(Qt::WA_DeleteOnClose);
-        connect(msgBox, &QDialog::finished, this, [this, signalSender](const int)
+        connect(msgBox, &QDialog::finished, this, [this, updater](const int)
```

Comment:
Seems it matches the connection above, unless diff confuses me. Can you avoid this duplication?

Reply:
I can add a lambda:
```c++
const auto cleanup = [this, updater](const int) { ... };
...
connect(msgBox, &QDialog::finished, this, cleanup);  // for dialog 1
...
connect(msgBox, &QDialog::finished, this, cleanup);  // for dialog 2
```
Would that be better?

Reply:
>I can add a lambda
Would that be better?

I think Yes.

Another way is create message box, connect common signals and then apply different properties (title, text, etc.) conditionally.